File: RaftSwiftGRPC/Sources/RaftCore/Utils/Client/Get.swift
/// Client to Leader, used for retrieving a value by key
public struct GetRequest: Sendable, Codable {
    /// Key to retrieve
    public let key: String

    /// Initializes a new GetRequest
    /// - Parameters:
    ///   - key: Key to retrieve
    public init(key: String) {
        self.key = key
    }
}

/// Leader to Client, response to GetRequest
public struct GetResponse: Sendable, Codable {
    /// Value associated with the key, null if not found
    public let value: String?

    /// Optional leader hint if this node is not the leader
    public let leaderHint: Peer?

    /// Initializes a new GetResponse
    /// - Parameters:
    ///   - value: Value associated with the key, null if not found
    ///   - leaderHint: Optional leader hint if this node is not the leader
    public init(value: String? = nil, leaderHint: Peer? = nil) {
        self.value = value
        self.leaderHint = leaderHint
    }
}

File: RaftKotlin/raft-node/src/main/kotlin/raft/core/utils/Client/Get.kt
package raft.core.utils.client

import raft.core.utils.types.Peer

/**
 * A request to get a value by key from the state machine.
 */
data class GetRequest(
    /**
     * Key to retrieve
     */
    val key: String
)

/**
 * A response to a get request.
 */
data class GetResponse(
    /**
     * Value associated with the key, null if not found
     */
    val value: String? = null,

    /**
     * Optional leader hint if this node is not the leader
     */
    val leaderHint: Peer? = null
)
